kafka-plugins/src/main/java/co/cask/hydrator/plugin/realtime/KafkaProducer.java

```diff
@@ -0,0 +1,283 @@
+    props.put(BROKER_LIST, sconfig.brokers);
+    props.put(KEY_SERIALIZER, "org.apache.kafka.common.serialization.StringSerializer");
+    props.put(VAL_SERIALIZER, "org.apache.kafka.common.serialization.StringSerializer");
+    props.put(CLIENT_ID, "kafka-producer-" + context.getInstanceId());
+    if (sconfig.isAsync.equalsIgnoreCase("TRUE")) {
+      props.put(ACKS_REQUIRED, "1");
+      isAsync = true;
+    }
+    
+    //config = new ProducerConfig(props);
+    producer = new org.apache.kafka.clients.producer.KafkaProducer<String, String>(props);
+    
+  }
+  
+  @Override
+  public int write(Iterable<StructuredRecord> objects, final DataWriter dataWriter) throws Exception {
+    int count = 0;
+
+    List<Schema.Field> fields = Lists.newArrayList();
+    
+    // For each object
+    for (StructuredRecord object : objects) {
+      
+      // Extract the field names from the object passed in. This is required
+      // because this information is not available in initialize or configuration phase.
+      if(!fieldsExtracted) {
+        fields = object.getSchema().getFields();
+        fieldsExtracted = true;
+      }
+      
+      // Depending on the configuration create a body that needs to be 
+      // built and pushed to Kafka. 
+      String body = "";
+      if (sconfig.format.equalsIgnoreCase("JSON")) {
+        body = StructuredRecordStringConverter.toJsonString(object);
+      } else {
+        // Extract all values from the structured record
+        List<Object> objs = Lists.newArrayList();
+        for(Schema.Field field : fields) {
+          objs.add(object.get(field.getName()));
+        }
+
+        StringWriter writer = new StringWriter();
+        CSVPrinter printer = null;
+        CSVFormat csvFileFormat;
+        switch(sconfig.format.toLowerCase()) {
+          case "csv":
+            csvFileFormat = CSVFormat.Predefined.Default.getFormat();
+            printer = new CSVPrinter(writer, csvFileFormat);
+            break;
+          
+          case "excel":
+            csvFileFormat = CSVFormat.Predefined.Excel.getFormat();
+            printer = new CSVPrinter(writer, csvFileFormat);
+            break;
+          
+          case "mysql":
+            csvFileFormat = CSVFormat.Predefined.MySQL.getFormat();
+            printer = new CSVPrinter(writer, csvFileFormat);
+            break;
+          
+          case "tdf":
+            csvFileFormat = CSVFormat.Predefined.TDF.getFormat();
+            printer = new CSVPrinter(writer, csvFileFormat);
+            break;
+          
+          case "rfc4180":
+            csvFileFormat = CSVFormat.Predefined.TDF.getFormat();
+            printer = new CSVPrinter(writer, csvFileFormat);
+            break;
+        }
+        
+        if (printer != null) {
+          printer.printRecord(objs);
+          body = writer.toString();
+        }
+      }
+      
+      // Message key.
+      String key = "no_key";
+      if(sconfig.key != null) {
```

Comment:
missing space after `if`. There are quite a number of them. Make sure they get fixed and better have checkstyle configured.